### PR TITLE
Cleanup output of e2e tests before generating XML

### PR DESCRIPTION
**Pull request by artemnikitin**
Because we are putting into file with test output stream from k8s , it's possible what something non-test related can go in like in https://devops-ci.elastic.co/view/cloud-on-k8s/job/cloud-on-k8s-versions-gke/86/execution/node/84/log/
```
14:42:38  gotestsum: Error: failed to parse test output: rpc error: code = Unknown desc = Error: No such container: 385257478d2fb705cb510d7e30f30640450cdcaa98ba2a021b0816eb4dc6202d: invalid character 'r' looking for beginning of value
14:42:38  Makefile:369: recipe for target 'e2e-generate-xml' failed
``` 
In this case an attempt to generate XML from test output will fail because output contains non-test related stuff.

Close https://github.com/elastic/cloud-on-k8s/issues/1967

### CODE REVIEW

**Comment by charith-elastic** [comment]
I wonder whether we should do this in the test runner itself. We can probably check whether the line is valid JSON before writing it to the file.

**Comment by artemnikitin** [comment]
> I wonder whether we should do this in the test runner itself. We can probably check whether the line is valid JSON before writing it to the file.

I would say if it's only for specific case (like right now) then in my opinion it's better to do it independently from test runner. In general, it's probably valuable to have all of output without any filtering

**Comment by artemnikitin** [comment]
@charith-elastic I implemented cleanup in test runner. Can you look at PR when you will have time please?

File: test/e2e/cmd/run/run.go [hunk 1]
```diff
@@ -484,3 +496,39 @@ func (h *helper) dumpK8sData() {
 		log.Error(err, "Failed to run hack/eck-dump.sh")
 	}
 }
+
+func (h *helper) cleanupLogFile() func() {
+	return func() {
+		log.Info("Cleaning up file with test log output")
+		content, err := ioutil.ReadFile(testsLogFile)
```
Comment (charith-elastic) [inline test/e2e/cmd/run/run.go]: I would use a `bufio.Reader` and call `ReadBytes('\n')` to avoid reading the entire file into memory at once.

Comment (artemnikitin) [reply test/e2e/cmd/run/run.go]: Fixed

**Comment by artemnikitin** [comment]
Closed in favor of https://github.com/elastic/cloud-on-k8s/pull/2095

**Pull request closed without merging**